frontend/src/components/resourceMap/useQueryParamsState.tsx

```diff
@@ -0,0 +1,43 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { useLocation } from 'react-router';
+
+type UseQueryParamsStateReturnType<T> = [T, Dispatch<SetStateAction<T>>];
+export function useQueryParamsState<T extends string | undefined>(
```

Comment:
This is a good candidate for extraction.

Reply:
moved to #2498